### ISSUE TITLE
Infinite Recursion when using !Ref within cloudformation template

### ISSUE THREAD

**Issue by CristianAThompson** [localstack/localstack#3438]
# Type of request: This is a ...

- [X] bug report
- [ ] feature request

# Detailed description
When using a cloudformation yml document to establish a queue within localstack a valid template that functions within AWS hits an infinite recursion issue when trying to create the resource locally that makes use of the !Ref intrinsic function to expose the URL for attaching this queue to another stack that establish a topic and subscriptions with subscription filters.
 
## Expected behavior
Using !Ref should properly generate the URL and allow it to be an exported value for this resource, not cause an infinite recursion that causes the process to die.

## Actual behavior
Infinite recursion/recursion limit reached with very simple resource declaration.

# Steps to reproduce

## Command used to start LocalStack
Localstack is configured as a docker container within a docker-compose.yml

#### Docker-Compose.yml localstack bit
```
version: '3'

services:
  localstack:
    image: localstack/localstack
    networks:
      static-network:
        ipv4_address: 172.18.0.6
    container_name: localstack_dev
    ports:
      - "4566-4597:4566-4597"
      - "8080:8080"
    environment:
      - SERVICES=sns,sqs,s3,cloudformation
      - PORT_WEB_UI=8080
      - DEBUG=1
      - DEFAULT_REGION=us-east-2
networks:
  static-network:
    ipam:
      config:
        - subnet: 172.18.0.1/24
```

#### First Queue Built - Our Central DL Queue ( Works fine )
```
# Stack Name: dead-letter-queue
Description: CloudFormation Template To Establish The Dead Letter Queue For All ModOn Queues 
Resources:
  ModOnDLQueue:
    Type: AWS::SQS::Queue
    Properties: 
      QueueName: ModOnDLQueue
Outputs:
  ModOnDLQueueArn:
    Description: The Dead Letter SQS Arn Value For Connecting To Other Stacks
    Value: !GetAtt ModOnDLQueue.Arn
    Export:
      Name: ModOnDLQueueArn
```
#### Second Queue Built - A Messaging Queue ( Infinite Recursion )
```
# Stack Name: message-queue
Description: CloudFormation Template To Establish Messaging SQS That Will Be Linked To Topics

Resources:
  MessageQueue:
    Type: AWS::SQS::Queue
    Properties: 
      QueueName: MessageQueue
      RedrivePolicy: 
        deadLetterTargetArn: !ImportValue ModOnDLQueueArn
        maxReceiveCount: 3
Outputs:
  MessageQueueArn:
    Description: The Message SQS Arn Value For Connecting To Other Stacks
    Value: !GetAtt MessageQueue.Arn
    Export:
      Name: MessageQueueArn
  MessageQueueUrl:
    Description: The Message SQS URL Value For Connecting To Other Stacks
    Value: !Ref MessageQueue
    Export:
      Name: MessageQueueUrl
```

#### Command executed to create both queues
```
aws --endpoint-url=http://localhost:4566 cloudformation create-stack --stack-name=dead-letter-queue --template-body=file://../path/to/yml/dead-letter-queue.yml && aws --endpoint-url=http://localhost:4566 cloudformation create-stack --stack-name=message-queue --template-body=file://../path/to/yml/message-queue.yml
```

#### Command to peek at localstack output
```
docker logs localstack_dev
```

#### Localstack Output
```
Waiting for all LocalStack services to be ready
2021-01-06 19:04:41,260 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2021-01-06 19:04:41,263 INFO supervisord started with pid 14
2021-01-06 19:04:42,266 INFO spawned: 'dashboard' with pid 20
2021-01-06 19:04:42,267 INFO spawned: 'infra' with pid 21
2021-01-06 19:04:42,272 INFO success: dashboard entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2021-01-06 19:04:42,272 INFO exited: dashboard (exit status 0; expected)
(. .venv/bin/activate; exec bin/localstack start --host)
Starting local dev environment. CTRL-C to quit.
LocalStack version: 0.12.4
2021-01-06 19:04:43,902 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Starting edge router (https port 4566)...
2021-01-06T19:04:45:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x7fb417877160>
Starting mock CloudFormation service on http port 4566 ...
Starting mock S3 service on http port 4566 ...
[2021-01-06 19:04:46 +0000] [22] [INFO] Running on https://0.0.0.0:4566 (CTRL + C to quit)
2021-01-06T19:04:46:INFO:hypercorn.error: Running on https://0.0.0.0:4566 (CTRL + C to quit)
2021-01-06T19:04:46:INFO:localstack.multiserver: Starting multi API server process on port 47565
[2021-01-06 19:04:46 +0000] [22] [INFO] Running on http://0.0.0.0:47565 (CTRL + C to quit)
2021-01-06T19:04:46:INFO:hypercorn.error: Running on http://0.0.0.0:47565 (CTRL + C to quit)
Starting mock SNS service on http port 4566 ...
Starting mock SQS service on http port 4566 ...
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
Ready.
2021-01-06T19:04:55:INFO:localstack.utils.analytics.profiler: Execution of "start_api_services" took 9256.792068481445ms
2021-01-06T19:05:38:DEBUG:localstack.services.cloudformation.cloudformation_api: Creating stack "dead-letter-queue" with 1 resources ...
2021-01-06T19:05:38:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "ModOnDLQueue" (1/1) type "AWS::SQS::Queue" in loop iteration 1
2021-01-06T19:05:38:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "SQS::Queue" id "ModOnDLQueue"
2021-01-06T19:05:38:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "SQS::Queue" in region us-east-2: create_queue {'QueueName': 'ModOnDLQueue', 'Attributes': {}, 'tags': {}}
2021-01-06 19:05:39,473 INFO exited: infra (exit status 2; not expected)
2021-01-06 19:05:39,474 INFO spawned: 'infra' with pid 94
(. .venv/bin/activate; exec bin/localstack start --host)
2021-01-06T19:05:39:DEBUG:localstack.services.cloudformation.cloudformation_api: Creating stack "message-queue" with 1 resources ...
2021-01-06T19:05:39:DEBUG:localstack.services.cloudformation.cloudformation_api: Unable to resolve references in stack outputs: {'Description': 'The Message SQS Arn Value For Connecting To Other Stacks', 'Value': {'Fn::GetAtt': ['MessageQueue', 'Arn']}, 'Export': {'Name': 'MessageQueueArn'}} - maximum recursion depth exceeded while calling a Python object
Fatal Python error: Cannot recover from stack overflow.
Python runtime state: initialized

Current thread 0x00007fb412e93b20 (most recent call first):
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 111 in get
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 570 in find_stack
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 626 in find_stack
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 993 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 943 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 185 in outputs
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 35 in exports
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 218 in exports_map
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 994 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 912 in resolve_ref
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 952 in resolve_refs_recursively
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1030 in resolve_refs_recursively
  ...

Thread 0x00007fb413099b20 (most recent call first):
  File "/usr/lib/python3.8/socket.py", line 669 in readinto
  File "/usr/lib/python3.8/http/client.py", line 264 in _read_status
  File "/usr/lib/python3.8/http/client.py", line 303 in begin
  File "/usr/lib/python3.8/http/client.py", line 1322 in getresponse
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 440 in _make_request
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699 in urlopen
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/requests/adapters.py", line 439 in send
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/requests/sessions.py", line 655 in send
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/requests/sessions.py", line 542 in request
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/requests/api.py", line 61 in request
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/requests/api.py", line 119 in post
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 407 in modify_and_forward
  File "/opt/code/localstack/localstack/services/edge.py", line 135 in do_forward_request_inmem
  File "/opt/code/localstack/localstack/services/edge.py", line 115 in do_forward_request
  File "/opt/code/localstack/localstack/services/edge.py", line 104 in forward_request
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 365 in modify_and_forward
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 597 in handler
  File "/opt/code/localstack/localstack/utils/async_utils.py", line 28 in _run
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 581 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007fb412f96b20 (most recent call first):
  File "/usr/lib/python3.8/selectors.py", line 415 in select
  File "/usr/lib/python3.8/socketserver.py", line 232 in serve_forever
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 777 in serve_forever
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 1009 in inner
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 1052 in run_simple
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/server.py", line 297 in main
  File "/opt/code/localstack/localstack/utils/server/multiserver.py", line 63 in thread_func
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 581 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007fb41319cb20 (most recent call first):
  File "/usr/lib/python3.8/selectors.py", line 468 in select
  File "/usr/lib/python3.8/asyncio/base_events.py", line 1823 in _run_once
  File "/usr/lib/python3.8/asyncio/base_events.py", line 570 in run_forever
  File "/usr/lib/python3.8/asyncio/base_events.py", line 603 in run_until_complete
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 157 in run_app_sync
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 183 in run_proxy
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 581 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007fb41329fb20 (most recent call first):
  File "/usr/lib/python3.8/selectors.py", line 415 in select
  File "/usr/lib/python3.8/socketserver.py", line 232 in serve_forever
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 777 in serve_forever
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 1009 in inner
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 1052 in run_simple
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 990 in run
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 632 in serve_flask_app
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 592 in serve
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 581 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007fb4133a2b20 (most recent call first):
  File "/usr/lib/python3.8/selectors.py", line 468 in select
  File "/usr/lib/python3.8/asyncio/base_events.py", line 1823 in _run_once
  File "/usr/lib/python3.8/asyncio/base_events.py", line 570 in run_forever
  File "/usr/lib/python3.8/asyncio/base_events.py", line 603 in run_until_complete
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 157 in run_app_sync
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 183 in run_proxy
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 581 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007fb413972b20 (most recent call first):
  File "/usr/lib/python3.8/threading.py", line 302 in wait
  File "/usr/lib/python3.8/queue.py", line 170 in get
  File "/opt/code/localstack/localstack/utils/analytics/event_publisher.py", line 138 in poll_and_send_messages
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 581 in run
  File "/usr/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007fb419e24d48 (most recent call first):
  File "/opt/code/localstack/localstack/utils/common.py", line 500 in sleep_forever
  File "/opt/code/localstack/localstack/services/infra.py", line 397 in start_infra
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 348 in start_infra_locally
  File "/opt/code/localstack/localstack/utils/cli.py", line 61 in cmd_infra
  File "/opt/code/localstack/localstack/utils/cli.py", line 183 in main
  File "bin/localstack", line 19 in <module>
make: *** [Makefile:54: infra] Aborted (core dumped)
Starting local dev environment. CTRL-C to quit.
LocalStack version: 0.12.4
2021-01-06 19:05:41,063 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Starting edge router (https port 4566)...
Starting mock CloudFormation service on http port 4566 ...
Starting mock S3 service on http port 4566 ...
2021-01-06T19:05:42:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x7f3663508160>
2021-01-06T19:05:42:INFO:localstack.multiserver: Starting multi API server process on port 37543
[2021-01-06 19:05:42 +0000] [95] [INFO] Running on https://0.0.0.0:4566 (CTRL + C to quit)
2021-01-06T19:05:42:INFO:hypercorn.error: Running on https://0.0.0.0:4566 (CTRL + C to quit)
[2021-01-06 19:05:42 +0000] [95] [INFO] Running on http://0.0.0.0:37543 (CTRL + C to quit)
2021-01-06T19:05:42:INFO:hypercorn.error: Running on http://0.0.0.0:37543 (CTRL + C to quit)
```

**Comment by CristianAThompson** [comment]
If we comment the MessageQueueUrl output the resource will then build correctly.

**Comment by CristianAThompson** [comment]
@whummer I appreciate you taking this issue on. I just wanted to check in to see what you think your priority on evaluating this will be? It's a blocker for our team as it stands, just asking so I can let our team know what to expect.

**Comment by CristianAThompson** [comment]
In doing some live editing of the docker container a spitting our references, I believe this section of code to be the culprit at first brush. I will continue digging into why this cyclical but as a first step it appears to occur here.

![image](https://user-images.githubusercontent.com/16628217/104054586-fcaa9b80-51b2-11eb-8c2a-2c755c00881e.png)

**Comment by whummer** [comment]
Thanks for reporting @CristianAThompson . This should (hopefully) be fixed in #3447 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks

**Comment by CristianAThompson** [comment]
That seems to have resolved that issue. I'm encountering another I will attempt to work around with some tooling. Appreciate you knocking this out so fast.
